src/dune/build.ml

```diff
@@ -118,7 +118,13 @@ let contents p = Contents p
 
 let lines_of p = Lines_of p
 
-let strings p = Map ((fun l -> List.map l ~f:Scanf.unescaped), lines_of p)
+let strings p =
+  let f x =
+    match Scanf.unescaped x with
+    | Error () -> Code_error.raise "Build.strings" []
```

Comment:
This is a file produced by the external world, so this shouldn't be a code error. What about raising a user error located in `p`?